alvr/audio/src/linux.rs

```diff
@@ -1,324 +1,255 @@
 use alvr_common::{ConnectionError, anyhow::Result, debug, error, parking_lot::Mutex};
 use alvr_session::AudioBufferingConfig;
 use alvr_sockets::{StreamReceiver, StreamSender};
+
+use std::os::unix::fs::FileTypeExt;
+use std::{
+    collections::VecDeque,
+    fs, io,
+    path::Path,
+    sync::{
+        Arc,
+        atomic::{AtomicBool, Ordering},
+    },
+    thread::{self, sleep},
+    time::Duration,
+};
+
 use pipewire::{
-    self as pw,
+    channel::Receiver,
+    context::Context,
+    core::Core,
+    keys,
+    main_loop::MainLoop,
+    properties,
     spa::{
-        self,
         param::audio::{AudioFormat, AudioInfoRaw},
-        pod::{self, Pod, serialize::PodSerializer},
+        pod::{self, Pod, Value, serialize::PodSerializer},
+        utils::Direction,
     },
-    stream::{StreamFlags, StreamListener, StreamState},
+    stream::{Stream, StreamFlags, StreamListener, StreamState},
 };
-use std::{cmp, collections::VecDeque, sync::Arc, thread, time::Duration};
+
+pub fn try_load_pipewire() -> Result<()> {
+    if let Err(e) = probe_pipewire() {
+        if !matches!(e, pipewire::Error::CreationFailed) {
+            return Err(e.into());
+        }
+        error!("Could not initialize PipeWire.");
+
+        let is_under_flatpak = std::env::var("FLATPAK_ID").is_ok();
+        let is_pw_socket_unavailable =
+            std::env::var("XDG_RUNTIME_DIR").is_ok_and(|xdg_runtime_dir| {
+                let pw_socket_path = Path::new(&xdg_runtime_dir).join("pipewire-0");
+                if pw_socket_path.exists()
+                    && let Ok(metadata) = fs::metadata(&pw_socket_path)
+                {
+                    !metadata.file_type().is_socket()
+                } else {
+                    false
```

Comment:
As a good example as why not to invert variables, it caused a bug here because it sets it as not unavailable aka available if the `socket_path` doesn't exist.

Reply:
thought inverting would make it better looking, but looks like it creates confusion if statement is simple, yeah...